scripts/test.sh

```diff
@@ -3,22 +3,22 @@ set -e
 
 # Brings in: ROOT_DIR, EXAMPLES_DIR, LIB_DIR, CODEGEN_DIR, CONTRIB_DIR, DOC_DIR
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-source $SCRIPT_DIR/config.sh
+source "${SCRIPT_DIR}/config.sh"
 
 # Add Cargo to PATH.
 export PATH=${HOME}/.cargo/bin:${PATH}
 
 # Checks that the versions for Cargo projects $@ all match
 function check_versions_match() {
   local last_version=""
-  for dir in $@; do
+  for dir in "$@"; do
```

Comment:
I believe we do need them, I did a quick check by creating a directory with spaces in the path like in the opening pull request description and without the double quotes it errors out looking for the wrong truncated path and doesn't error with the quotes.